operations/repo.go

```diff
@@ -203,3 +260,104 @@ func buildRepo(ctx context.Context, packages, configPath, workingDir, distro, ed
 
 	return errors.Wrapf(catcher.Resolve(), "encountered problem rebuilding repository after %d retries", retries)
 }
+
+type barqueServiceInfo struct {
+	url      string
+	username string
+	password string
+}
+
+func submitRepo(ctx context.Context, info barqueServiceInfo, configPath, distro, edition, version, arch string, packages []string) error {
+	// validate inputs
+	if edition == "community" {
+		edition = "org"
+	}
+
+	// get configuration objects.
+	conf, err := repobuilder.GetConfig(configPath)
+	if err != nil {
+		grip.Error(err)
+		return errors.Wrap(err, "problem getting repo config")
+	}
+
+	repo, ok := conf.GetRepositoryDefinition(distro, edition)
+	if !ok {
+		e := fmt.Sprintf("repo not defined for distro=%s, edition=%s ", distro, edition)
+		grip.Error(e)
+		return errors.New(e)
+	}
+
+	opts := repobuilder.JobOptions{
+		Configuration: conf,
+		Distro:        repo,
+		Version:       version,
+		Arch:          arch,
+		Packages:      packages,
+		JobID:         "tk",
+	}
+
+	client, err := barquesubmit.New(info.url)
+	if err != nil {
+		return errors.Wrap(err, "problem constructing barque client")
+	}
+
+	if err = client.Login(ctx, info.username, info.password); err != nil {
+		return errors.Wrap(err, "problem authenticating to barque")
+	}
+
+	id, err := client.SubmitJob(ctx, opts)
+	if err != nil {
+		return errors.Wrap(err, "problem submitting repobuilder job")
+	}
+
+	startAt := time.Now()
+	checks := 0
+	timer := time.NewTimer(15 * time.Second)
+	defer timer.Stop()
+RETRY:
+	for {
+		checks++
+		select {
+		case <-ctx.Done():
+			return errors.New("operation timed out")
+		case <-timer.C:
+			stat, err := client.CheckJobStatus(ctx, id)
+			if err != nil {
+				grip.Error(err)
+				return errors.Wrap(err, "problem checking job status")
+			}
+
+			if !stat.Status.Completed {
+				grip.Info(message.Fields{
+					"job":               stat.ID,
+					"wallclock_seconds": time.Since(startAt).Seconds(),
+					"duration_seconds":  time.Since(stat.Timing.Start).Seconds(),
+					"in_progress":       stat.Status.InProgress,
+					"checks":            checks,
+				})
+				timer.Reset(30*time.Second + time.Duration(rand.Int63n(int64(time.Minute))))
```

Comment:
Just out of curiosity, what' the point of adding the jitter?

Reply:
avoiding different clients hitting the server in lockstep?